addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -23,33 +23,47 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
         return lines
 
     def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
+        if self._context.get('pricelist') != self.pricelist_id.id:
+            self = self.with_context(pricelist=self.pricelist_id.id)
```

Comment:
Hmmm.

That introduces context management that will go up to the super method.
Is that intended? Seems a bit louche.